### PR TITLE
Fix/nng enable stats

### PR DESCRIPTION
**Pull request by robiwano**
Fixes some compilation issues when NNG_ENABLE_STATS is set to OFF.

### CODE REVIEW

**Comment by robiwano** [comment]
Also adds nng_http_server_get_addr symbol to the CMakeLists.txt file for doc generation.

File: src/core/dialer.c [hunk 1]
```diff
@@ -128,6 +128,9 @@ dialer_stats_init(nni_dialer *d)
 void
 nni_dialer_bump_error(nni_dialer *d, int err)
 {
+#ifndef NNG_ENABLE_STATS
+	NNI_ARG_UNUSED(d);
```
Comment (gdamore) [inline src/core/dialer.c]: Probably should have done this for err too.

File: src/core/listener.c [hunk 2]
```diff
@@ -125,6 +125,9 @@ listener_stats_init(nni_listener *l)
 void
 nni_listener_bump_error(nni_listener *l, int err)
 {
+#ifndef NNG_ENABLE_STATS
```
Comment (gdamore) [inline src/core/listener.c]: Also for err.

**Comment by robiwano** [comment]
Hmm... one test is failing, not sure why. Investigating.

**Comment by gdamore** [comment]
I don't know either.  The old Convey tests are hard to debug, and I've found them somewhat fragile in some of the cloud providers.  I'm in the process of converting them.  At any rate, I will waive the test failure, since it clearly isn't in the code you changed.

**Pull request merged**